### ISSUE TITLE
Prevent add-ons which are "disabled" from interacting with NVDA at all.

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#10133]
### Is your feature request related to a problem? Please describe.
Recently I encountered a bug which turned out to be caused by a misbehaving add-on.  The add-on in question was installed but "Disabled".

If an add-on is disabled, the user expectation is that it cannot do anything or cause any behaviour within NVDA.

### Describe the solution you'd like

To bring NVDA into line with that expectation and stop add-ons from having any effect when they are disabled.

### Describe alternatives you've considered

If an add-on is no longer needed, it should be uninstalled, but often it is desirable to disable an add-on for a time, either to use alternate functionality or when testing problems.  When testing problems particular, it is expected that disabling add-ons should resolve any issues caused by misbehaving add-ons but this is not always the case.

**Comment by josephsl** [comment]
Hi, can you provide us with the name and the symptom from this misbehaving add-on so authors can fix it with help from users? Thanks.

**Comment by Qchristensen** [comment]
The specific instance was focus highlight as per #10131 - but we have had reports from users who have had addons disabled and they've still caused problems (possibly focus highlight, possibly others).

As noted in the issue, my suggestion is not around fixing an individual add-on (although that should happen as well, ideally), but that this shouldn't be able to happen in the first place.  As a user, if I disable an add-on - in any program - I expect that means nothing in the add-on is running, and it doesn't have any effect on the program until I re-enable it.  I think that is a reasonable expectation.

**Comment by Brian1Gaff** [comment]
Yes I'd always assumed this was the case, ie it is seen as an add on but 
when its disabled no further action other than reporting its details should 
be needed. Is this sloppy programming or would this need new guidelines in 
add on  writing docs?

I mean somebody may define things used for the about  etc, in some variables 
but accidentally do some other bits without realising it.
  Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
-----

**Comment by JulienCochuyt** [comment]
Could you please provide us with a reproducible scenario with a disabled add-on having any side effect?

As stated in https://github.com/nvaccess/nvda/issues/10131#issuecomment-526662049, I can't figure out how/why focusHighlight would be the culprit in #10123

**Comment by lukaszgo1** [comment]
@Qchristensen Could you please respond to [this commend](https://github.com/nvaccess/nvda/issues/10133#issuecomment-526813470) by @JulienCochuyt  ?

**Comment by lukaszgo1** [comment]
@Qchristensen Ping. Have you got any reproducible scenario in which disabled add-on is still interfering with NVDA functionality?

**Comment by Qchristensen** [comment]
I must have been having a vague month last August!  I don't have a reproducible scenario for this, so I'm going to close this one.  If anyone does encounter anything and can write it up better than I did this one, it might be easiest to just create a new issue :)
